src/components/Slider/SliderHooks.js

```diff
@@ -0,0 +1,83 @@
+import { useEffect, useRef, useState } from "react";
+
+function ensureSingleValueText(valueText, value, formatter) {
+  if (valueText) {
+    return valueText;
+  }
+  if (typeof value === "undefined") {
+    return undefined;
+  }
+  if (typeof formatter !== "function") {
+    return value;
+  }
+  return formatter(value);
+}
+
+function ensureValueText(valueText, value, formatter) {
+  if (!Array.isArray(value)) {
+    return ensureSingleValueText(valueText, value, formatter);
+  }
+  return value.map((valueSingle, index) => {
+    const valueTextSingle = Array.isArray(valueText) ? valueText[index] : undefined;
+    return ensureSingleValueText(valueTextSingle, valueSingle, formatter);
+  });
+}
+
+export function useControlledOrInternal(value) {
```

Comment:
suggestion for simpler function name: "useIsStateControlledFromOutside" 
the current name sounds like it will return an enum

Reply:
Done!